sstables/sstables.cc

```diff
@@ -3191,6 +3212,9 @@ future<> init_metrics() {
         sm::make_counter("total_deleted", [] { return sstables_stats::get_shard_stats().deleted; },
             sm::description("Counter of deleted sstables")),
 
+        sm::make_gauge("bytes_on_disk", [] { return sstables_stats::get_shard_stats().bytes_on_disk; },
+            sm::description("Bytes currently used on disk by sstables")),
+
```

Comment:
We could consider making this part of sstables_manager, or adding a label to support different storage classes. Could be added later. /cc @xemul

Reply:
> We could consider making this part of sstables_manager

@raphaelsc indicated something similar in the original bug report #18191, but (a) I found stats being set up in a bunch of -- apparently -- unrelated locations and functions already, and (b) I figured if it could be kept within the `sstable` class, then why would we want to drag `sstables_manager` into it? Isolation of responsibilites etc.

> or adding a label to support different storage classes

I don't understand this. What are labels? Thanks.

(I assume by "storage classes" you mean filesystem storage vs S3 storage.)

Reply:
Metrics labels. A metric can be tagged with a set of labels and then each unique set of label instances becomes its own time series. If we had a label storage_class=object|local then we could count object storage separately from local storage (and sum them if wanted). We always have implicit labels on the shard and instance.

Reply:
done in v3